pkgs/development/tools/godot/default.nix

```diff
@@ -1,103 +1,146 @@
-{ stdenv
-, lib
+{ alsa-lib
 , fetchFromGitHub
-, scons
-, pkg-config
-, udev
+, freetype
+, installShellFiles
+, lib
+, libGLU
+, libpulseaudio
 , libX11
 , libXcursor
+, libXext
+, libXfixes
+, libXi
 , libXinerama
 , libXrandr
 , libXrender
-, libpulseaudio
-, libXi
-, libXext
-, libXfixes
-, freetype
 , openssl
-, alsa-lib
-, libGLU
-, zlib
+, pkg-config
+, scons
+, stdenv
+, udev
 , yasm
-, withUdev ? true
+, zlib
 }:
 
-let
-  options = {
-    touch = libXi != null;
-    pulseaudio = false;
-    udev = withUdev;
-  };
-in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (self: {
   pname = "godot";
   version = "3.4.4";
 
   src = fetchFromGitHub {
     owner = "godotengine";
     repo = "godot";
-    rev = "${version}-stable";
-    sha256 = "sha256-3AESLzqozi7Fc80u8Ml3ergZMkIhHy4tNlRe/3FsE6k=";
+    rev = self.passthru.githubRev;
+    sha256 = self.passthru.githubSha256;
+  };
+
+  passthru = {
+    githubRev = "${self.version}-stable";
+    githubSha256 = "sha256-3AESLzqozi7Fc80u8Ml3ergZMkIhHy4tNlRe/3FsE6k=";
+
+    sconsFlagSet = {
+      target = "release_debug";
+      platform = "x11";
+      tools = true;
+      touch = libXi != null;
```

Comment:
libXi is a package, right? Then it should never be null.

Reply:
I didn't write the `touch = libXi != null` logic; I just kept it in my refactor.  Looking into it, though, I see no reason for that attribute to ever be set to null.  I'll just change that to `true`.